libbeat/management/config.go

```diff
@@ -0,0 +1,72 @@
+package management
+
+import (
+	"os"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/file"
+	"github.com/elastic/beats/libbeat/paths"
+	"github.com/elastic/beats/libbeat/setup/kibana"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Config for central management
+type Config struct {
+	// true when enrolled
+	Enabled bool
+
+	// TODO use beat.Keystore() for access_token
+	AccessToken string
+
+	Kibana *kibana.Config
+
+	Configs []struct {
+		Name   string
+		Config *common.Config
+	}
+}
+
+// Load settings from its source file
+func (c *Config) Load() error {
+	path := paths.Resolve(paths.Data, "management.yaml")
+	config, err := common.LoadFile(path)
+	if err != nil {
+		return err
+	}
+
+	if err = config.Unpack(c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Save settings to management.yaml file
+func (c *Config) Save() error {
+	path := paths.Resolve(paths.Data, "management.yaml")
+
+	data, err := yaml.Marshal(c)
+	if err != nil {
+		return err
+	}
+
+	// write temporary file first
+	// TODO this should be owned by the beats user, ensure that
```

Comment:
I remember we even have some reusable code for this to make sure it's the correct user.